product_restricted_type/models/product_category.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2018-2024 ForgeFlow S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductCategory(models.Model):
+    _inherit = "product.category"
+
+    restricted_product_type = fields.Selection(
```

Comment:
@AnaisLForgeFlow For the next time, try to limit your changes in migration commit to strictly migration ones. The better is to put improvements in separate ones.